paddle/fluid/framework/tensor.h

```diff
@@ -37,23 +41,51 @@ class Tensor {
 #ifdef PADDLE_WITH_MKLDNN
 
  public:
-  inline mkldnn::memory::format format() const { return format_; }
+  // TODO(jczaja): This is depracted and will be removed
+  inline mkldnn::memory::format format() const {
+    if (layout_ == DataLayout::kMKLDNN) {
+      return static_cast<mkldnn::memory::format>(mem_pd_.desc().data.format);
+    } else {
+      return mkldnn::memory::format::format_undef;
+    }
+  }
+
+  // TODO(jczaja): This is depracted and will be removed
+  inline void set_format(
+      const mkldnn::memory::format fmt,
+      mkldnn::memory::data_type data_type = mkldnn::memory::f32) {
+    mem_pd_ = paddle::platform::create_prim_desc_from_format(
+        paddle::framework::vectorize2int(dims()), fmt, data_type);
+    layout_ = DataLayout::kMKLDNN;
+  }
 
-  inline void set_format(const mkldnn::memory::format format) {
-    format_ = format;
+  inline mkldnn::memory::primitive_desc get_prim_desc() const {
```

Comment:
this method and the ones above should be called `mkldnn_prim_desc` to make its usage more obvious.

Reply:
@panyx0718 Ok, methods renamed as suggested.